ReactAndroid/src/main/java/com/facebook/react/devsupport/DevSupportManagerImpl.java

```diff
@@ -921,9 +918,39 @@ public void onServerContentChanged() {
         mDevOptionsDialog.dismiss();
       }
 
+      // hide loading view
+      if (mDevLoadingView != null) {
+        mDevLoadingView.hide();
+      }
+
       mDevServerHelper.closePackagerConnection();
       mDevServerHelper.closeInspectorConnection();
       mDevServerHelper.stopPollingOnChangeEndpoint();
     }
   }
+
+  private void hideDevLoadingView() {
+    if (mDevLoadingView != null && mDevLoadingViewVisible) {
+      mDevLoadingView.hide();
+      mDevLoadingViewVisible = false;
+    }
+  }
+
+  private DevLoadingView getDevLoadingView() {
+    if (mDevLoadingView == null) {
+      mDevLoadingView = new DevLoadingView(mApplicationContext);
+      WindowManager windowManager = (WindowManager) mApplicationContext.getSystemService(Context.WINDOW_SERVICE);
+
+      WindowManager.LayoutParams params = new WindowManager.LayoutParams(
+        WindowManager.LayoutParams.MATCH_PARENT,
+        (int) PixelUtil.toPixelFromDIP(22),
```

Comment:
Yea I could use just the text + some padding instead of centering with fixed height.